src/emcpy/plots/create_plots.py

```diff
@@ -194,8 +197,10 @@ def __init__(self, figsize=(8, 6)):
 
         super().__init__(figsize)
         self.ax = self.fig.add_subplot(111)
+        self.shared_ax = None
+        self.shared_ay = None
 
-    def draw_data(self, plot_list):
+    def draw_data(self, plot_list, use_shared_ax=False, use_shared_ay=False):
```

Comment:
My general comment is: can we just use `self.shared_ax` and not add these extra function/method arguments? They should be stored in `obj` or `plotobj`, right? Or maybe we need `self.use_shared_ax`, or something like that?

Reply:
Yes, I think that would work.  I'll give it a try.  Thanks.

Reply:
Second cut.  The axes (default ax, optional shared_ax, shared_ay) are properties of the CreatePlot class.  The default, ax, is created when the object is instantiated, the shared axes as needed. 

The plot objects now have a plot_ax variable set to 'ax', 'shared_ax', or 'shared_ay' ('ax' is default).  To do that I added a BasePlot class to emcpy/plots/plots.py which all the other classes now extend.  It's the BasePlot that includes the methods to change the value of plot_ax.  This means the calls to plotting is much more Python-like -- just instantiate all your plot objects, set the desired axes and plot all in one call.

@CoryMartin-NOAA , @kevindougherty-noaa Please take a look when you have a moment.  I still have to work out labels for the second x/y axis, and I have to figure out how to get the legend to work for those figures on shared axes.  I'll hold on that though pending your feedback on this approach.  Thanks.

Reply:
From what I saw, this looks great to me, @EdwardSafford-NOAA . The `BasePlot` class is a good addition, and I like the approach of the `_get_axis` call, very neat and clean.

Reply:
Super.  It took me a while to think more OO than procedural; it's been a long time but glad to be back in OO land.  I'll figure out the axis labels for other than the default axes next, and then clean it all up for a proper PR.